aviary/mission/problem_builder_height_energy.py

```diff
@@ -0,0 +1,443 @@
+        )
+
+    def get_phase_builder(self, prob, phase_name, phase_options):
+
+        if 'phase_builder' in phase_options:
+            phase_builder = phase_options['phase_builder']
+            if not issubclass(phase_builder, PhaseBuilderBase):
+                raise TypeError(f"phase_builder for the phase called "
+                                "{phase_name} must be a PhaseBuilderBase object.")
+        else:
+            phase_builder = EnergyPhase
+
+        return phase_builder
+
+    def set_phase_options(self, prob, phase_name, phase_idx, phase, user_options):
+
+        try:
+            fix_initial = user_options.get_val('fix_initial')
+        except KeyError:
+            fix_initial = False
+
+        try:
+            fix_duration = user_options.get_val('fix_duration')
+        except KeyError:
+            fix_duration = False
+
+        # The rest of the phases includes all Height Energy method phases
+        # and any 2DOF phases that don't fall into the naming patterns
+        # above.
+        input_initial = False
+        time_units = phase.time_options['units']
+
+        # Make a good guess for a reasonable intitial time scaler.
+        try:
+            initial_bounds = user_options.get_val('initial_bounds', units=time_units)
+        except KeyError:
+            initial_bounds = (None, None)
+
+        if initial_bounds[0] is not None and initial_bounds[1] != 0.0:
+            # Upper bound is good for a ref.
+            user_options.set_val('initial_ref', initial_bounds[1],
+                                 units=time_units)
+        else:
+            user_options.set_val('initial_ref', 600., time_units)
+
+        duration_bounds = user_options.get_val("duration_bounds", time_units)
+        user_options.set_val(
+            'duration_ref', (duration_bounds[0] + duration_bounds[1]) / 2.,
+            time_units
+        )
+        if phase_idx > 0:
+            input_initial = True
+
+        if fix_initial or input_initial:
+
+            if prob.comm.size > 1:
+                # Phases are disconnected to run in parallel, so initial ref is
+                # valid.
+                initial_ref = user_options.get_val("initial_ref", time_units)
+            else:
+                # Redundant on a fixed input; raises a warning if specified.
+                initial_ref = None
+
+            phase.set_time_options(
+                fix_initial=fix_initial, fix_duration=fix_duration, units=time_units,
+                duration_bounds=user_options.get_val("duration_bounds", time_units),
+                duration_ref=user_options.get_val("duration_ref", time_units),
+                initial_ref=initial_ref,
+            )
+        elif phase_name == 'descent':  # TODO: generalize this logic for all phases
+            phase.set_time_options(
+                fix_initial=False, fix_duration=False, units=time_units,
+                duration_bounds=user_options.get_val("duration_bounds", time_units),
```

Comment:
The only difference between this elif and the code before it is that if the phase_name is 'descent' then fix_duration is set to False, no matter what the user input. That is a check that should be happening on the phase inputs, not here. That would allow us to delete lines 158 through 165.

Honestly a bit strange to me that we specifically need fix_duration = False in descent.

Reply:
Yeah, that is strange too, plus it is the only example where we have to adhere to a naming convention in HE.  It looks like fix_duration is always false in our phase_infos, and the ones in MBSAE.  If the user sets `target_duration` in a phase, then fix_duration is automatically set to True (we have so much interaction between these attributes), but in those cases, the phase is named something like "reserve_hold" instead of "descent".  I am going to remove this "if" branch and see what happens.